File: WhatsAppClone/src/main/java/com/lnreddy/WhatsAppClone/notification/model/Notification.java
package com.lnreddy.WhatsAppClone.notification.model;

import com.lnreddy.WhatsAppClone.message.constants.MessageState;
import com.lnreddy.WhatsAppClone.message.constants.MessageType;
import com.lnreddy.WhatsAppClone.notification.constants.NotificationType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Notification {

    private UUID chatId;
    private UUID messageId;
    private UUID senderId;
    private String content;
    private MessageType messageType;
    private NotificationType notificationType; // MESSAGE, IMAGE, SEEN
    private UUID userId; // who saw it
    private List<UUID> seenBy;
    private byte[] media;

}

File: whatsapp-clone-ui/src/app/pages/main/notification.ts
export interface Notification{

    chatId?:string;
    content?: string;
    senderId?: string;
    messageId?: string;
    receiverIds?: Array<string>;
    messageType?: 'TEXT'| 'IMAGE' |  'AUDIO' | 'VIDEO',
    notificationType?: 'SEEN' | 'MESSAGE' | 'IMAGE' | 'VIDEO' | 'AUDIO';
    media?: Array<string>
   
}